src/EnergyPlus/ScheduleManager.cc

```diff
@@ -1641,354 +1636,229 @@ namespace ScheduleManager {
         if (NumCommaFileSchedules > 0) {
             hourlyFileValues.allocate(8784 * 60); // sized to accomodate any interval for schedule file.
         }
+
         CurrentModuleObject = "Schedule:File";
-        for (LoopIndex = 1; LoopIndex <= NumCommaFileSchedules; ++LoopIndex) {
-            state.dataInputProcessing->inputProcessor->getObjectItem(state,
-                                                                     CurrentModuleObject,
-                                                                     LoopIndex,
-                                                                     Alphas,
-                                                                     NumAlphas,
-                                                                     Numbers,
-                                                                     NumNumbers,
-                                                                     Status,
-                                                                     lNumericBlanks,
-                                                                     lAlphaBlanks,
-                                                                     cAlphaFields,
-                                                                     cNumericFields);
-            GlobalNames::VerifyUniqueInterObjectName(
-                state, state.dataScheduleMgr->UniqueScheduleNames, Alphas(1), CurrentModuleObject, cAlphaFields(1), ErrorsFound);
-            ++SchNum;
-            state.dataScheduleMgr->Schedule(SchNum).Name = Alphas(1);
-            state.dataScheduleMgr->Schedule(SchNum).SchType = SchedType::ScheduleInput_file;
-            // Validate ScheduleType
-            if (state.dataScheduleMgr->NumScheduleTypes > 0) {
-                CheckIndex = 0;
-                if (!lAlphaBlanks(2))
-                    CheckIndex =
-                        UtilityRoutines::FindItemInList(Alphas(2), state.dataScheduleMgr->ScheduleType({1, state.dataScheduleMgr->NumScheduleTypes}));
-                if (CheckIndex == 0) {
-                    if (!lAlphaBlanks(2)) {
-                        ShowWarningError(state,
-                                         "ProcessScheduleInput: For " + CurrentModuleObject + "=\"" + Alphas(1) + "\", " + cAlphaFields(2) + "=\"" +
-                                             Alphas(2) + "\" not found -- will not be validated");
-                    } else {
-                        ShowWarningError(state,
-                                         "For " + CurrentModuleObject + "=\"" + Alphas(1) + "\", Blank " + cAlphaFields(2) +
-                                             " input -- will not be validated.");
-                    }
-                } else {
-                    state.dataScheduleMgr->Schedule(SchNum).ScheduleTypePtr = CheckIndex;
+
+        // Runs getObjectItem for Schedule:File and saves it to vector in state: allIdfSchedData
+        PreProcessIDF(state, SchNum, NumCommaFileSchedules);
```

Comment:
I wrapped the getObjectItem function and all the variable assignments into this PreProcessIDF function. This will populate the allIdfSchedData vector. We can access every Schedule:File object data from here, since these have been saved as elements of the allIdfSchedData vector. The main module also looks cleaner.